documentation/v&v/V&V.tex

```diff
@@ -0,0 +1,588 @@
+	\item The application will be accessible from any device with a modern browser (phone, tablet, desktop, etc.) to provide the best experience.
+\ee
+
+
+
+\section{Background}
+\label{sec:background}
+University students often like to buy/sell used textbooks. Sometimes it's for the notes written in the margins, but often it's because new textbooks are quite expensive.\\
+Currently at McMaster University, students use online websites like Facebook (\url{https://www.facebook.com/groups/macusedtexts}) or the campus store's buyback program (\url{https://campusstore.mcmaster.ca/information/guaranteed-buyback.html}, suspended because of COVID) to sell their books. These current solutions are either disorganized or take a cut of the transaction. Book Bazaar aims to create an organized marketplace that allows users to sell books for a better price than at the campus store.\\
+The project, Book Bazar is a website that addresses these issues. McMaster community members may buy/sell textbooks with ease using Book Bazar.
+\section{Roadmap}
+\label{sec: roadmap}
+Using the milestones, listed in Section 6 of the Development Process document, the team aims to have Book Bazar completed by the week of March 6, a week prior to the team's f\textbf{}inal presentation.
+
+\section{Testing Plan}
+Integration testing was used for all top-level modules which transitively tests all bottom level modules. This style of testing was chosen because it effectively tests how users will interact with the system instead of implementation details. Unit tests were not used as in practice they are often burdensome to write and maintain while providing minimal benefit in many cases due to much of the system being mocked out.
+
+\subsection{Testing Tools}
+The team will be use the following tools for implementing tests:\\
+\bi
+    \item Cypress -- used to run all integration tests (API and UI).
+    \item Jest -- used for writing unit tests.
+    \item Mock Service Workers -- used for mocking out external service APIs.
+\ei
+\subsection{API Tests}
+In the design specification the behaviour of HTTP handler modules was defined. This behaviour was used to define an expected output for a variety of inputs. This required creating tooling to build and teardown a testing database.
+
+A serverless architecture was used for API execution which means that horizontal scaling of the system is handled by \href{https://vercel.com}{Vercel}, the hosting provider. Therefore, \textbf{load testing} was not performed.
+
+
+\subsection{UI Tests}
+UI tests have been created for each way a user can interact with a page.
+
+
+\subsection{PASS/FAIL Criterion}
+The criterion used to determine PASS/FAIL of the the test cases id given below:
+\begin{center}
+    {$actual\_output == expected\_output$}
+\end{center}
+
+
+
+\section{Traceability Matrix}
+Below tables go over the coverage of our test plan. Modules were tested multiple times and ensured that the Function Requirements implemented are correct. 
```

Comment:
This table is a traceability matrix, not a coverage report.

"Function Requirements" ==> "Functional Requirements"

Drop the phrasing "are correct". It's impossible to prove that a program is correct through testing. Instead, use the wording "build confidence that the implementation matches the functional requirements"